internal/server/server.go

```diff
@@ -71,6 +71,7 @@ func New(cfg config.Config) (*Server, error) {
 	}
 
 	s.logger.Info("Initializing server...")
+	s.logger.Info("Initializing database proxy listener...", logr.String("listen-address", cfg.ListenAddress))
```

Comment:
Nit: Maybe move `logr.String("listen-address", cfg.ListenAddress)` as part of the first log line itself? "server" and "database proxy listener" kinda mean the same thing.